pkg/iptables/iptables.go

```diff
@@ -36,103 +71,128 @@ func NewTrafficRedirector(
 	tr *TrafficRedirectionSpec,
 	executor runtime.Executor,
 ) (protocol.TrafficRedirector, error) {
-	if tr.DestinationPort == 0 || tr.RedirectPort == 0 {
-		return nil, fmt.Errorf("the DestinationPort and RedirectPort must be specified")
+	if tr.TargetPort == "" || tr.ProxyPort == "" {
+		return nil, fmt.Errorf("TargetPort and ProxyPort must be specified")
 	}
 
-	if tr.DestinationPort == tr.RedirectPort {
-		return nil, fmt.Errorf(
-			"the DestinationPort (%d) and RedirectPort (%d) must be different",
-			tr.DestinationPort,
-			tr.RedirectPort,
-		)
+	if tr.TargetPort == tr.ProxyPort {
+		return nil, fmt.Errorf("TargetPort (%s) and ProxyPort (%s) must be different", tr.TargetPort, tr.ProxyPort)
 	}
 
-	if tr.Iface == "" {
-		return nil, fmt.Errorf("the Iface must be specified")
+	if tr.LocalAddress == "" || tr.Interface == "" {
+		return nil, fmt.Errorf("local address and interface must be specified")
 	}
 
 	return &redirector{
 		TrafficRedirectionSpec: tr,
 		executor:               executor,
+		ip:                     iproute.New(executor),
 	}, nil
 }
 
 // delete iptables rules for redirection
 func (tr *redirector) deleteRedirectRules() error {
-	return tr.execRedirectCmd("-D")
+	for _, chain := range redirectChains {
+		if err := tr.execRedirectCmd("-D", chain); err != nil {
+			return err
+		}
+	}
+
+	return nil
```

Comment:
This logic should be rewritten so an error deleting the first rule does not prevent the next one from being deleted.